File: app/XDM/XDM.Core/MediaParser/Hls/HlsMediaSegment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDM.Core.MediaParser.Hls
{
    public class HlsMediaSegment
    {
        public HlsMediaSegment(Uri url)
        {
            this.Url = url;
        }
        public Uri Url { get; set; }
        public KeyValuePair<long, long> ByteRange { get; set; }
        public double Duration { get; set; }
        public Uri? KeyUrl { get; set; }
        public string? IV { get; set; }
    }
}

File: xdm-core/src/main/java/xdm/core/media/parser/hls/HlsMediaSegment.java
package xdm.core.media.parser.hls;

import static java.util.AbstractMap.SimpleEntry;

import java.net.URI;
import lombok.Builder;
import lombok.Data;
import lombok.ToString;

@Builder
@Data
@ToString
public class HlsMediaSegment {
  private URI url;
  private SimpleEntry<Long, Long> byteRange;
  private double duration;
  private URI keyUrl;
  private String iv;
}
